eth/catalyst/api_types.go

```diff
@@ -102,9 +92,28 @@ type ConsensusValidatedParams struct {
 	Status    string      `json:"status"`
 }
 
+// PayloadID is an identifier of the payload build process
+type PayloadID [8]byte
+
+func (b PayloadID) String() string {
+	return hexutil.Encode(b[:])
+}
+
+func (b PayloadID) MarshalText() ([]byte, error) {
+	return hexutil.Bytes(b[:]).MarshalText()
+}
+
+func (b *PayloadID) UnmarshalText(input []byte) error {
+	err := hexutil.UnmarshalFixedText("PayloadID", input, b[:])
+	if err != nil {
+		return fmt.Errorf("invalid payload id %q: %w", input, err)
+	}
+	return nil
+}
```

Comment:
Any particular reason for making this type instead of just using a `uint8` for example? Or `hexutil.Bytes`? Or if you really want decode time protection, perhaps add `hexutil.Uint8` analogously to `hexutil.Uint64`?

Reply:
The PayloadID type? It's 8 bytes, so a uint64 would work, but it's defined as `DATA`. Other lengths are incorrect. For the `LogsBloom` I would honestly also like to see decode-time protection. It helps catch bugs as an engine user if it doesn't quietly ignore malformatted / missing data.

Reply:
Ah I see, yes, that makes it a bit clearer why you chose this approach.